webapp/routes/route_team.jsx

```diff
@@ -147,6 +152,63 @@ function onPermalinkEnter(nextState, replace, callback) {
     );
 }
 
+function onChannelByIdentifierEnter(state, replace, callback) {
+    if (state.params.identifier.indexOf('@') === -1) {
+        handleError(state, replace, callback);
+    } else {
+        let username = state.params.identifier;
+        if (username.indexOf('@') === 0) {
+            username = username.slice(1, username.length);
+        } else if (username.indexOf('@') > 0) {
+            username = username.slice(0, username.indexOf('@'));
```

Comment:
@hmhealey adding a `UserStore.getProfileByEmail(email);` would mean also adding  to mattermost-redux repo a `getUserByEmail` https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/users.js. Is this something you could add? I don't think there's a proper way to do it on platform directly.